heiseonline/shariff
# Add new service Pocket

This pull request (PR) adds the [Pocket](https://getpocket.com/) service to Shariff.

Solution for issue #152 .

Thanks to [https://github.com/vinkla/shareable-links/blob/master/README.md#pocket](https://github.com/vinkla/shareable-links/blob/master/README.md#pocket) for the sharing URL and parameters.

## Important note for maintainers:

@compeak When PR #348 will be merged, this PR here will have conflicts.

I suggest you merge PR #348 , then I solve conflicts in this PR here, then you merge this PR here and make a new release.

Another thing: Pocket is not really for sharing but for saving. I would understand if you decide not to merge this PR here because Shariff focuses on sharing. On the other hand we already have a link for printing, and you can say that saving to Pocket is like sharing with yourself but on different devices, and the heiseonline articles use already a button for Pocket, see sharing buttons e.g. on [https://heise.de/-4266139](https://heise.de/-4266139) at the right side at bottom of the article text:
![heiseonline-pocket-button](https://user-images.githubusercontent.com/7413183/50736815-6a85ec00-11c2-11e9-86a9-c777bb264cfe.png)
I leave it up to you, but I think that in opposite to e.g. the "Like" button of Facebook or some "Follow" buttons, a Pocket button would be within the functional focus of Shariff.

## How to test:

Run the shariff demo site locally on a Linux host which has git and npm installed with the branch of this PR as follows:

```sh
$ mkdir shariff-test-pocket
$ cd shariff-test-pocket
$ git clone https://github.com/richard67/shariff-plus.git
$ cd shariff-plus
$ git checkout upstream-add-pocket
$ git pull origin upstream-add-pocket
$ npm install
$ npm run dev
```
Check the result in the browser, URL=`http://localhost:3000/`.

#### Check the result

Verify that a new button for Pocket has been added at the end of examples 2, 3 and 4 of the demo, that the icons and colors fit to the branding on [https://getpocket.com/](https://getpocket.com/) and that the sharing links work and use the right title and URL.

#### Clean up after the test
```sh
$ cd ../..
$ rm -rf ./shariff-test-pocket
```
